enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+cluster_installation_modification{asset="cluster-scheduler-02-config.yml"} 1
+```
+The potential values for these labels would be the ignition files: `bootstrap.ign`, `master.ign`, or `worker.ign` or the manifest files:
+```
+04-openshift-machine-config-operator.yaml  cluster-network-01-crd.yml       etcd-client-secret.yaml                etcd-namespace.yaml                 machine-config-server-tls-secret.yaml
+cloud-provider-config.yaml                 cluster-network-02-config.yml    etcd-host-service-endpoints.yaml       etcd-service.yaml                   openshift-config-secret-pull-secret.yaml
+cluster-config.yaml                        cluster-proxy-01-config.yaml     etcd-host-service.yaml                 etcd-serving-ca-configmap.yaml
+cluster-dns-02-config.yml                  cluster-scheduler-02-config.yml  etcd-metric-client-secret.yaml         etcd-signer-secret.yaml
+cluster-infrastructure-02-config.yml       cvo-overrides.yaml               etcd-metric-serving-ca-configmap.yaml  kube-cloud-config.yaml
+cluster-ingress-02-config.yml              etcd-ca-bundle-configmap.yaml    etcd-metric-signer-secret.yaml         kube-system-configmap-root-ca.yaml
+```
+
+The number of potential values is approximately 30.
+
+Either of these proposed metrics would be pushed only upon successful completion of the `create cluster` command.
+
+#### Metric Labels
+Collecting labels into a single metric sample and pushing once means that each sample represents a single installer invocation. When a sample is pushed to the aggregation pushgateway metrics with matching labels will be incremented:
+
+```
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="25"} 100
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="GCP",result="APIFailure",return="1",version="4.3",duration="30"} 25
+cluster_installation_invocation{command="destroy", target="cluster",os="linux",platform="AWS",result="Success",return="1",version="4.2",duration="10"} 5
+```
+If the above example represents the state of our Prometheus database, it would indicate that there have been 100 successful cluster installations on AWS that took 25 or fewer minutes.
+
+The values for these labels may only become available at different times during an installation or command execution. Therefore the installer instrumentation must handle building labels by collecting key value pairs as they become available.
+
+#### Internal Installer Design
+A new metrics package would be created to handle the collection of metrics, building labels, and pushing to the aggregation pushgateway. 
+
+As required label values are gathered throughout many installer packages,  the `metrics` package would maintain state package-scoped variables (feedback please: package-scoped variables may be generally frowned upon but is this a valid use case?).
```

Comment:
i think it's important that the package allow initiating a sink for specific purpose.

- but we can start with all metrics being collected into the global sink.
- an alternative would be initiate a sink per new asset store to reduce use of global state.